### PR TITLE
Pass killGracePeriodSeconds value to Mesos for pods

### PR DESCRIPTION
**Pull request by timcharper**
JIRA Issues: MARATHON-7952

### CODE REVIEW

**Comment by kensipe** [comment]
nice work in short time frame

File: src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala [hunk 1]
```diff
@@ -183,6 +185,15 @@ object TaskGroupBuilder extends StrictLogging {
       computeHealthCheck(healthCheck, portAssignments).foreach(builder.setHealthCheck)
     }
 
+    for {
+      lc <- container.lifecycle
+      killGracePeriodSeconds <- lc.killGracePeriodSeconds
+    } {
+      val durationInfo = DurationInfo.newBuilder.setNanoseconds((killGracePeriodSeconds * 1000000000).toLong)
```
Comment (jeschkies) [inline src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala]: Nit: I like to have Java do the transformation `Duration.ofSeconds(killGracePeriodSeconds).getNano()`. This avoids magic numbers.

File: src/test/scala/mesosphere/mesos/TaskGroupBuilderTest.scala [hunk 2]
```diff
@@ -1055,5 +1056,29 @@ class TaskGroupBuilderTest extends UnitTest with Inside {
...
+      val (_, taskGroupInfo, _, _) = TaskGroupBuilder.build(
+        podSpec,
+        offer,
+        s => Instance.Id.forRunSpec(s),
+        defaultBuilderConfig,
+        RunSpecTaskProcessor.empty,
+        resourceMatch.asInstanceOf[ResourceMatchResponse.Match].resourceMatch
+      )
+
+      taskGroupInfo.getTasks(0).getKillPolicy.getGracePeriod.getNanoseconds shouldBe (killDuration.toNanos)
```
Comment (jeschkies) [inline src/test/scala/mesosphere/mesos/TaskGroupBuilderTest.scala]: Here you use `toNanos` 😉

File: src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala (hunk 1 above)
Comment (timcharper) [reply src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala]: Yeah, I thought about this and anticipated somebody might bring this up. I totally prefer to use durations. In this case, the input duration was a double and already unit-less. Didn't seem worth the extra allocation. And the conversion was covered by a test.

**Comment by kensipe** [comment]
@timcharper while not the norm... I would support a merge.    @jeschkies rejection was addressed and I wouldn't want to wait all weekend to get another review.     There isn't a "rush" except for the customer that needs this... needs a back port.

**Pull request merged**